lib/jsdom/browser/Window.js

```diff
@@ -403,6 +404,21 @@ function matchesDontThrow(el, selector) {
 }
 
 function startTimer(window, startFn, stopFn, callback, ms) {
+  if (typeof callback !== "function") {
+    const code = String(callback);
+    callback = window._globalProxy.eval.bind(window,
+      `(function(){` + code + `}())\n//# sourceURL=${window.location.href}`);
```

Comment:
Does this sourceURL actually work in node? Or is this purely for browser users?

It seems better to use vm since we can specify the filename there.